python/webhook/cascade-status-change/status_change/app.py

```diff
@@ -0,0 +1,184 @@
+# :coding: utf-8
+# :copyright: Copyright (c) 2018 ftrack
+import json
+
+import functools
+import logging
+import os
+
+import ftrack_api
+
+import logging
+logger = logging.getLogger()
+logger.setLevel("INFO")
+
+ALLOWED_STATES = ('BLOCKED', 'DONE', 'IN_PROGRESS', 'NOT_STARTED')
+
+SUCCESS_RESPONSE = {
+    "statusCode": 200,
+    "body": json.dumps({
+        "message": "SUCCESS",
+    }),
+}
+
+
+ERROR_RESPONSE = {
+    "statusCode": 400,
+    "body": json.dumps({
+        "error": "An error occurred during the lambda execution. Please check logs",
+    }),
+}
+
+
+def send_message_to_user(session, user_id):
+    '''Send a success message to the active user.
+
+    Use the event hub of *session* to pop up a message for the user with
+    *user_id*. (Functionality new in ftrack 3.3.31.)
+    '''
+    session.event_hub.publish(
+        ftrack_api.event.base.Event(
+            topic='ftrack.action.trigger-user-interface',
+            data=dict(
+                type='message',
+                success=True,
+                message=(
+                    'cascade_status_changes: ' 'Shot status updated automatically'
+                ),
+            ),
+            target=f'applicationId=ftrack.client.web and user.id="{user_id}"'
+        ),
+        on_error='ignore',
+    )
+
+def get_status_by_state(project, state):
+    '''Return a valid Status which matches *state*, for the given *project*.
+
+    Raise an exception if the Shot Schema for *project* has no Status with the
+    given *state*.
+    '''
+    for status in project['project_schema'].get_statuses('Shot'):
+        if status['state']['short'] == state:
+            return status
+
+    raise ValueError(
+        f'No valid Shot status matching state {state} for project { project["full_name"]}'
+    )
+
+
+def is_status_change(event):
```

Comment:
This method will return True for any update to a task, not status changes only. Consider renaming the method or checking if the change was a status change and not something else (preferred).